server/services/area.service.js

```diff
@@ -12,3 +12,16 @@ export const addArea = async (area) => {
     throw new ErrorHandler(httpStatus.BAD_REQUEST, 'Error adding area', error);
   }
 };
+
+export const getStateAndArea = async (state) => {
+  if (!state) {
+    const states = await Area.distinct('state');
+    return { states };
+  }
+  const areas = await Area.find({ state });
```

Comment:
I believe the below lines are not needed since we don't know the format the frontend will need. It is better to just return the id and area as an array and let the frontend convert it to the needed format. 

```suggestion
  const areas = await Area.find({ state }).select('area');
  return { areas };
```